drivers/wifi/nrf700x/osal/fw_if/umac_if/src/fmac_api.c

```diff
@@ -2952,6 +2952,82 @@ enum wifi_nrf_status wifi_nrf_fmac_set_power_save(void *dev_ctx,
 }
 
 
+enum wifi_nrf_status wifi_nrf_fmac_get_power_save_config(void *dev_ctx,
+							 unsigned char if_idx,
+							 struct wifi_ps_config *ps_cfg)
+{
+	enum wifi_nrf_status status = WIFI_NRF_STATUS_FAIL;
+	struct nrf_wifi_umac_cmd_get_power_save_cfg *get_ps_cfg_cmd = NULL;
+	struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx = NULL;
+
+	fmac_dev_ctx = dev_ctx;
+
+	get_ps_cfg = wifi_nrf_osal_mem_zalloc(fmac_dev_ctx->fpriv->opriv,
+					      sizeof(*get_ps_cfg));
+
+	if (!get_ps_cfg) {
+		wifi_nrf_osal_log_err(fmac_dev_ctx->fpriv->opriv,
+				      "%s: Unable to allocate memory\n", __func__);
+		goto out;
+	}
+
+	get_ps_cfg_cmd->umac_hdr.cmd_evnt = NRF_WIFI_UMAC_CMD_GET_POWER_SAVE_CFG;
+	get_ps_cfg_cmd->umac_hdr.ids.wdev_id = if_idx;
+	get_ps_cfg_cmd->umac_hdr.ids.valid_fields |=
+		NRF_WIFI_INDEX_IDS_WDEV_ID_VALID;
+	get_ps_cfg_cmd->info.ps_state = state;
+
+	status = umac_cmd_cfg(fmac_dev_ctx,
+			      get_ps_cfg_cmd,
+			      sizeof(*get_ps_cfg_cmd));
+out:
+	if (get_ps_cfg_cmd) {
+		wifi_nrf_osal_mem_free(fmac_dev_ctx->fpriv->opriv,
+				       get_ps_cfg_cmd);
+	}
+
+	return status;
+}
+
+
+enum wifi_nrf_status wifi_nrf_fmac_set_power_save_mode(void *dev_ctx,
+						       unsigned char if_idx,
+						       int mode)
+{
+	enum wifi_nrf_status status = WIFI_NRF_STATUS_FAIL;
+	struct nrf_wifi_umac_cmd_set_power_save_mode *set_ps_mode_cmd = NULL;
+	struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx = NULL;
+
+	fmac_dev_ctx = dev_ctx;
+
+	set_ps_mode_cmd = wifi_nrf_osal_mem_zalloc(fmac_dev_ctx->fpriv->opriv,
+					      sizeof(*set_ps_cmd));
+
+	if (!set_ps_mode_cmd) {
+		wifi_nrf_osal_log_err(fmac_dev_ctx->fpriv->opriv,
+				      "%s: Unable to allocate memory\n", __func__);
```

Comment:
Generic comment: Please put each parameter on a separate line (i.e. in this case __func__ can be moved to the next line).